turbopack/crates/turbo-tasks-fs/src/lib.rs

```diff
@@ -1116,7 +1116,7 @@ impl FileSystemPath {
     pub async fn join(self: Vc<Self>, path: RcStr) -> Result<Vc<Self>> {
         let this = self.await?;
         if let Some(path) = join_path(&this.path, &path) {
-            Ok(Self::new_normalized(this.fs, path.into()))
+            Ok(Self::new_normalized(*this.fs, path.into()))
```

Comment:
Should we make the public signature of turbo-tasks functions take `impl Into<Vc<...>>` instead of `Vc<...>`, so one can pass Vc and ResolvedVc (and VcOperation)?

Reply:
That's an interesting idea that I haven't considered. I'm going to be conservative and say "no" for now because:

- This deref is probably the easiest thing for me to codemod. In most cases the compiler can auto-suggest the exact change needed.
- Making the functions generic might have a slight negative impact on code size and compile times.
- This more complicated signature might make the rustdocs harder to read?

But if the derefs end up being annoying to people (added syntactic noise), we can do this.

Regarding "magic" in API (and especially macro) design: I'd like to keep it to a minimum. I didn't really like the change I had to do to let functions accept `ResolvedVc` exposed as `Vc` types, but I had to do it because the consequences of not doing that were far worse.